unified-runtime/source/adapters/level_zero/v2/command_buffer.cpp

```diff
@@ -37,7 +200,77 @@ ur_exp_command_buffer_handle_t_::ur_exp_command_buffer_handle_t_(
           context, device,
           std::forward<v2::raii::command_list_unique_handle>(commandList),
           v2::EVENT_FLAGS_COUNTER, nullptr),
-      isUpdatable(desc ? desc->isUpdatable : false) {}
+      isUpdatable(desc ? desc->isUpdatable : false), context(context),
+      device(device) {}
+
+ur_result_t ur_exp_command_buffer_handle_t_::updateKernelHandle(
+    locked<ur_command_list_manager> &commandListLocked,
+    ur_kernel_handle_t newKernel, kernel_command_handle *command) {
+  ze_kernel_handle_t kernelHandle = newKernel->getZeHandle(device);
+  auto platform = context->getPlatform();
+  auto commandId = command->commandId;
+  ZE2UR_CALL(
+      platform->ZeMutableCmdListExt
+          .zexCommandListUpdateMutableCommandKernelsExp,
+      (commandListLocked->getZeCommandList(), 1, &commandId, &kernelHandle));
+  // Set current kernel to be the new kernel
+  command->kernel = newKernel;
+  return UR_RESULT_SUCCESS;
+}
+ur_result_t ur_exp_command_buffer_handle_t_::createCommandHandle(
```

Comment:
Function looks very similar to `createCommandHandle` in v1 adapter, can we share code?